### ISSUE TITLE
Taiga Upgrade Broken (Uses Wrong Credentials)

### ISSUE THREAD

**Issue by muellert** [taigaio/taiga-back#1298]
**Before adding an issue**
Please ensure that your issue is not already reported in our [issues list](https://tree.taiga.io/project/taiga/issues?order_by=-created_date).
If this issue was already reported, remember that you can upvote it to raise its importance.

**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Upgrade does not work

**If the current behavior is a bug, please provide the steps to reproduce.**

Start from a 4.0 tree, try to upgrade to 4.2.3 according to the instructions on the upgrade page found here:

https://taigaio.github.io/taiga-doc/dist/upgrades.html

At this point:

$ python manage.py migrate --noinput

the upgrade process fails with an error message about the local user, who owns the Taiga instance, not being a role in the database (yes - the database user is something completely unrelated in my installation).

The result is a long traceback, which ends in

  File "/opt/taiga/backend/.virtualenvs/taiga/lib/python3.5/site-packages/psycopg2/__init__.py", line 130, in connect                                                            
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)                                                                                                            
django.db.utils.OperationalError: FATAL:  role "mytaigauser" does not exist

**What is the expected behavior?**

Upgrade works.

The upgrade process should take all the database settings and credentials from settings/local.py instead of assuming that the database is local, and that it can connect via Unix domain socket.

**Is it happening in taiga.io or in your own instance?**

It happens in my own instance.

**What browser/version are you using?**

Not relevant - this is a backend problem.

**Are there any console errors *(Ctrl + F12)* in red?**

No.

Suggested fix:

in settings/development.py, add this directly after "from .common import *":

from .local import *

**Comment by alexhermida** [comment]
@muellert currently it's working well. Maybe I didn't understand the issue. 

For migrating database, you must have configured your database settings correctly in the file `settings/local.py`

Something like:

```
DATABASES = {
    "default": {
        "ENGINE": "django.db.backends.postgresql",
        "NAME": "database_name",
        "USER": "database_username",
        "PASSWORD": "database_password",
        "HOST": "database_host",
        "PORT": database_port,
    }
}
```
With this configuration, your local or remote connection works well. It's the way that Taiga is usually deployed.

Regarding the error, it's some PostgreSQL misconfiguration. It seems you are connecting to a database without that role. Check if you have any other PostgreSQL service running.

**Comment by muellert** [comment]
As I said, I had to add 'from .local import *' to development.py before it would work. The setup worked in every other way before and after the upgrade. But without this line, the upgrade would break and try to use the default database setting, because development.py imports from .common.py, which has an empty setting for the database, and is loaded after local.py, thus overwriting the (correct) database settings in local.py.

**Comment by alexhermida** [comment]
Currently `common.py` has a [default database setting](https://github.com/taigaio/taiga-back/blob/master/settings/common.py#L32). You only have to configure the `local.py` with your custom settings.

In your production environment, you shouldn't use the settings from `development.py`. That file is though for a local development environment. You must remove the line `from .development import *` from the `local.py` file. 

Maybe we should add a comment to that line at the `local.py.example` to clarify that it only makes sense in development environments.

**Comment by theriverman** [comment]
> Maybe we should add a comment to that line at the `local.py.example` to clarify that it only makes sense in development environments.

It can be considered, but the documentation for production deployment clearly states to create a `local.py` file and configure it. Any installation with tampered `common.py` or missing `local.py` is a deviation from official instructions.

**Comment by muellert** [comment]
I am not sure what you guys are talking about. I have upgraded my Taiga instance several times, and it usually worked without a hitch. Just during this upgrade, it broke in the way I said. I haven't changed anything from my end until it broke. In local.py, I don't import from development.py, but when I ran the migration command, as outlined above, I could see in the output that development.py was somehow imported. I didn't track where that got imported, though, but found that IF development.py is being imported after local.py, it will overwrite the database setting, resulting in the breakage which I described.
